### PR TITLE
cli: get-file-locations --verbose option

### LINKED ISSUE #83: cli: `get-file-locations --verbose`
Currently, we output only file locations when users use the `get-file-locations` command:

```console
$ cernopendata-client get-file-locations --recid 1 | head -3
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/00E16FBB-9071-E011-83D3-003048673F12.root
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/0248915F-EE71-E011-8894-0025902009E8.root
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/0268F635-B671-E011-9090-002481E14E00.root
```

This does not inform users about the size or the checksum of what is going to be uploaded.

This information could be useful to estimate the download times and/or to plug this into automated scripts the users may have.

In order to improve the situation, we can introduce a new option `--verbose`  (or perhaps two new options `--include-size` and `--include-chechksum`?) that would also print  out the file size and the checksum information. 

Example:

```console
$ cernopendata-client get-file-locations --recid 1 | head -3
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/00E16FBB-9071-E011-83D3-003048673F12.root 123234234234 adler32:aaaaa
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/0248915F-EE71-E011-8894-0025902009E8.root 234234324 adler32:bbbbb
http://opendata.cern.ch/eos/opendata/cms/Run2010B/BTau/AOD/Apr21ReReco-v1/0000/0268F635-B671-E011-9090-002481E14E00.root 67676767676 adler32:ccccc
```

I.e. the output would be space-separated URI SIZE CHECKSUM triad.

CC @jbenito3

### LINKED ISSUE #92: cernopendata-client download-files --recid 3005 --verify --protocol https
```
$ cernopendata-client download-files --recid 3005 --verify --protocol https
==> Downloading file 1 of 1
  -> File: ./3005/0d0714743f0204ed3c0144941e6ce248.configFile.py
Traceback (most recent call last):
  File "/home/simko/.virtualenvs/cernopendata-client/bin/cernopendata-client", line 8, in <module>
    sys.exit(cernopendata_client())
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/cernopendata_client/cli.py", line 291, in download_files
    download_single_file(path=path, file_location=file_location, protocol=protocol)
  File "/home/simko/.virtualenvs/cernopendata-client/lib/python3.8/site-packages/cernopendata_client/downloader.py", line 68, in download_single_file
    c.perform()
pycurl.error: (1, 'Protocol "root" not supported or disabled in libcurl')
```

### PR DESCRIPTION
**Pull request by ParthS007**
closes #83
closes #92

### CODE REVIEW

File: cernopendata_client/cli.py [hunk 1]
```diff
@@ -125,7 +125,13 @@ def get_metadata(server, recid, doi, title, output_value):
     type=click.STRING,
     help="Which CERN Open Data server to query? [default={}]".format(SERVER_HTTP_URI),
 )
-def get_file_locations(server, recid, doi, title, protocol, expand):
+@click.option(
+    "--verbose",
+    is_flag=True,
+    default=False,
+    help="Output the file size and the checksum information.",
```
Comment (tiborsimko) [inline cernopendata_client/cli.py:132]: Output also the file size (in the second column) and the file checksum (in the third column).

File: docs/usage.rst [hunk 2]
```diff
@@ -133,6 +133,26 @@ would rather see the equivalent XRootD endpoints for the files:
 
 The data files can be downloaded via XRootD protocol using the **xrdcp** command.
 
+**Verbose**
```
Comment (tiborsimko) [inline docs/usage.rst:136]: Instead of "Verbose", you can perhaps name it "File sizes and checksums"

File: docs/usage.rst [hunk 3]
```diff
@@ -133,6 +133,26 @@ would rather see the equivalent XRootD endpoints for the files:
...
+
+Note that you can use ``--verbose`` command-line option if you would rather want to see
```
Comment (tiborsimko) [inline docs/usage.rst:138]: Let's revert the order:

If you would like to know in advance the file sizes and checksums, you can use `--verbose` option:

**Review by tiborsimko — COMMENTED** [review]
Works nicely :+1: Left a few cosmetic suggestions

File: docs/usage.rst [hunk 4]
```diff
@@ -204,6 +204,24 @@ The default timeout is 60 seconds.
     CMS_Run2010B_BTau_AOD_Apr21ReReco-v1_0005_file_index.json
     CMS_Run2010B_BTau_AOD_Apr21ReReco-v1_0005_file_index.txt
     ..
+**File sizes and checksums**
```
Comment (tiborsimko) [inline docs/usage.rst:207]: docs/usage.rst:207:Explicit markup ends without a blank line; unexpected unindent.

Comment (tiborsimko) [reply docs/usage.rst:208]: BTW I'm seeing interesting failures locally:

```
__________________________________________________________________ test_download_files_https_pycurl __________________________________________________________________

    def test_download_files_https_pycurl():
        """Test download_files() command with https protocol using pycurl."""
        pycurl = pytest.importorskip("pycurl")  # noqa: F841
        test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
        if os.path.isfile(test_file):
            os.remove(test_file)
        test_download_files = CliRunner()
        test_result = test_download_files.invoke(
            download_files, ["--recid", 3005, "--protocol", "https"]
        )
>       assert test_result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result error(1, 'Protocol "root" not supported or disabled in libcurl')>.exit_code

tests/test_cli_download_files.py:99: AssertionError
_________________________________________________________________ test_download_files_https_requests _________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x7f3c388bf790>

    def test_download_files_https_requests(mocker):
        """Test download_files() command with https protocol using requests."""
        mocker.patch("cernopendata_client.downloader.pycurl_available", False)
        test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
        if os.path.isfile(test_file):
            os.remove(test_file)
        test_download_files = CliRunner()
        test_result = test_download_files.invoke(
            download_files, ["--recid", 3005, "--protocol", "https"]
        )
>       assert test_result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result InvalidSchema("No connection adapters were found for 'root://eospublic.cern.ch//eos/opendata/cms/configuration-files/2011/0d0714743f0204ed3c0144941e6ce248.configFile.py'")>.exit_code

tests/test_cli_download_files.py:117: AssertionError
```

Shouldn't `root` be gone, and why are invoking XRootD in `test_download_files_https_requests()`?

Comment (ParthS007) [reply docs/usage.rst:208]: I fixed it, I had it in mind to replace the `root` with `https` in `get_files_list` after the `https` PR get merged.

Comment (tiborsimko) [reply docs/usage.rst:208]: Seeing this one now:

```
________________________________________________________________________ test_validate_server ________________________________________________________________________

    def test_validate_server():
        """Test validate_server()."""
        assert validate_server("http://opendata.cern.ch") is True
        assert validate_server("http://opendata-dev.cern.ch") is True
        assert validate_server("http://0.0.0.0:5000") is True
>       pytest.raises(SystemExit, validate_server, "https://opendata.cern.ch")
E       Failed: DID NOT RAISE <class 'SystemExit'>

tests/test_validator.py:36: Failed
```

Does it pass for you locally?

Comment (ParthS007) [reply docs/usage.rst:208]: Yes 😅 , Have to fix the test for `https` as well.

Comment (tiborsimko) [reply docs/usage.rst:208]: We can merge this one and address the rest in #92. It could be a bigger change, I left some comments there.

**Review by tiborsimko — APPROVED** [review]
The tests pass locally:

```
=== 63 passed in 29.40s ===
```

**Pull request merged**